rfcs/0001-fobnail-architecture-change.md

```diff
@@ -1,143 +1,207 @@
+VPN.
 
 It is also not possible to expose any services from Fobnail Token to Attester,
 such as access to symmetric cryptographic keys or performing cryptographic
-operations on behalf of attester in case of asymmetric cryptography. This is a
-must for features like disk encryption or Fobnail-based authentication to
-services, like VPN.
+operations on behalf of Attester in case of asymmetric cryptography. Such an API
+is a must for features like disk encryption or Fobnail-based authentication to
+services like VPN.
 
-Switching to a server based Fobnail Token allows to expose services for
+Switching to a server-based Fobnail Token allows exposing services for
 accessing cryptographic keys and performing administrative tasks.
 
 # Architecture change proposal
 
 ## Fobnail Token provisioning
 
-Currently, if in unprovisioned state, Fobnail Token initiates provisioning by
-requesting PO certificate chain from Platform Owner, then it verifies PO
+Currently, if in an unprovisioned state, Fobnail Token initiates provisioning by
+requesting PO certificate chain from the Platform Owner and verifies PO
 certificate chain against a trusted certificate embedded into Fobnail firmware.
 If verification passes, Fobnail generates Identity/Encryption keypair and
-requests PO to generate certificate from this key. Retrieval of valid
+requests PO to generate Identity/Encryption certificate. Retrieval of valid
 Identity/Encryption certificate finalizes provisioning process.
 
-Instead we propose the following protocol:
-1. Unprovisioned Fobnail token awaits for server to commence provisioning by
-   sending it's certificate chain.
+Instead, we propose the following protocol:
+1. Unprovisioned Fobnail Token awaits for client (Platform Owner) to commence
+   provisioning by sending its certificate chain.
 2. Fobnail verifies certificate chain against trusted certificate embedded into
    its firmware. If verification is succesful, Fobnail generates
-   Identity/Encryption keypair and responds with Certificate Signing Request.
-   Otherwise an error is returned.
+   Identity/Encryption keypair and responds with a Certificate Signing Request.
 3. At this point Platform Owner configures Fobnail, like default appraisal
-   policy. This step is optional, if skipped, defaults will be used.
-4. Platform finalizes configuration by sending certificate generated from
+   policy. This step is optional and if skipped, defaults will be used.
+4. Platform Owner finalizes configuration by sending certificate generated from
    previously provided CSR.
 5. Fobnail Token verifies and saves certificate and provided configuration to
    its internal memory, then locks configuration and sends status to Platform
    Owner, finalizing provisioning process.
 6. Token can be unlocked only by reset, which wipes out all configuration,
    provisioned platforms and all keys.
 
+![](img/remote-platform-provisioning.svg)
```

Comment:
@arturkow2000 would you mind to copy legend as it is in local provisioning?

Reply:
done